### PR TITLE
AP_Mount: support gimbal_manager_status and do_gimbal_manager_configure

### PR DESCRIPTION
**Pull request by rmackay9**
This supersedes PR https://github.com/ArduPilot/ardupilot/pull/23713

This adds superficial support for two new mavlink messages meant to help de-conflict multiple mavlink devices attempting to control a gimbal.  This is important so that development of [QGC's camera/gimbal control screen](https://github.com/mavlink/qgroundcontrol/pull/10667) can progress.

- [GIMBAL_MANAGER_STATUS](https://github.com/ArduPilot/mavlink/blob/master/message_definitions/v1.0/common.xml#L5895) reports to the GCS which device is currently in control of the gimbal (e.g. sysid, compid)
- [MAV_CMD_DO_GIMBAL_MANAGER_CONFIGURE](https://github.com/ArduPilot/mavlink/blob/master/message_definitions/v1.0/common.xml#L1625) allows a GCS to take or give-up control of the gimbal

The support is "superficial" because it does not actually reject gimbal control commands that comes from devices that are not in control.  To comprehensively block commands we need to consolidate handling of more commands (including DO_SET_ROI) within the AP_Mount library instead of within GCS_MAVLink.

This has been lightly tested in SITL including:

- GIMBAL_MANAGER_STATUS message is returned in response to "long MAV_CMD_REQUEST_MESSAGE 281"
- MAV_CMD_DO_GIMBAL_MANAGER_CONFIGURE commands are consumed and correctly affect the sysid/compid returned in the above message especially when the special values are passed in for sysid (e.g. 0:no one in control, -1:leave unchanged, -2:set itself in control, -3:remove control if currently in control)

testing-output

There's also a drive-by bug fix to the [GIMBAL_MANAGER_INFORMATION](https://github.com/ArduPilot/mavlink/blob/master/message_definitions/v1.0/common.xml#L5881) gimbal_device_id instance field.

### CODE REVIEW

**Review by MichelleRos — APPROVED** [review]
All good for the blimp changes - sorry, I really need to do a lot of removal of unused code there...

**Comment by Davidsastresas** [comment]
Looking great, thank you!

File: libraries/AP_Mount/AP_Mount_Backend.cpp [hunk 1]
```diff
@@ -267,6 +286,42 @@ MAV_RESULT AP_Mount_Backend::handle_command_do_mount_control(const mavlink_comma
...
+// requires original message in order to extract caller's sysid and compid
+MAV_RESULT AP_Mount_Backend::handle_command_do_gimbal_manager_configure(const mavlink_command_long_t &packet, const mavlink_message_t &msg)
+{
+    // sanity check param1 and param2 values
+    if ((packet.param1 < -3) || (packet.param1 > UINT8_MAX) || (packet.param2 < -3) || (packet.param2 > UINT8_MAX)) {
+        return MAV_RESULT_FAILED;
+    }
+
+    // convert negative packet1 and packet2 values
+    int16_t new_sysid = packet.param1;
```
Comment (peterbarker) [inline libraries/AP_Mount/AP_Mount_Backend.cpp:299]:
```suggestion
    const int16_t new_sysid = packet.param1;
```

File: libraries/AP_Mount/AP_Mount_Backend.cpp [hunk 2]
```diff
@@ -162,7 +162,7 @@ void AP_Mount_Backend::send_gimbal_manager_information(mavlink_channel_t chan)
     mavlink_msg_gimbal_manager_information_send(chan,
                                                 AP_HAL::millis(),                       // autopilot system time
                                                 get_gimbal_manager_capability_flags(),  // bitmap of gimbal manager capability flags
-                                                _instance,                              // gimbal device id
+                                                _instance + 1,                          // gimbal device id
```
Comment (tridge) [inline libraries/AP_Mount/AP_Mount_Backend.cpp:165]: can you check with David Sastre on this change? maybe add comment to xml if this is right

Comment (rmackay9) [reply libraries/AP_Mount/AP_Mount_Backend.cpp:165]: [We've discussed in this discussion](https://github.com/mavlink/mavlink/pull/1980) and we've agreed it should be 1 based.

File: libraries/AP_Mount/AP_Mount_Backend.cpp (hunk 1 above)
Comment (rmackay9) [reply libraries/AP_Mount/AP_Mount_Backend.cpp:299]: Txs.  I'll leave this const out for now but at the next opportunity..

**Comment by trolledbypro** [comment]
Hello all,
I am working on increasing MAVSDK compatibility with ArduPilot. My issue on their repo is https://github.com/mavlink/MAVSDK/issues/2110. Through testing, I have discovered that MAVSDK sends DO_GIMBAL_MANAGER_CONFIGURE as part of the mission sequence.

Is this PR intended to allow DO_GIMBAL_MANAGER_CONFIGURE to work as a mission item, sent inside a mission sequence?

**Pull request merged**